### ISSUE TITLE
[Proposal: trace_attributes] add "traffic_signal" to "edges"

### ISSUE THREAD

**Issue by upperwal** [valhalla/valhalla#1932]
I ***propose*** to export/add a new field `traffic_signal` of type `bool` in `edges` object. 

This field will signify if an edge has a traffic signal or not. There could be two cases:

1. Traffic signal is somewhere within the edge:
`---x---->`
2. Signal is at the end node (`intersection`):
`--------x`

Both these cases will yield:
```json
{
  "edges": [
    {
      "speed": 50,
      "length": 0.079,
      "traffic_signal": true
    }
  ]
}
```

I would be happy to open a draft pull request if the community things it can benefit them. Do let me know. @kevinkreiser @dgearhart

**Comment by upperwal** [comment]
@kevinkreiser would love to know your views on this. I already have an implementation and could open a PR for the same.

**Comment by kevinkreiser** [comment]
@upperwal the traffic signal happens at the nodes which makes it a bit odd on the edge. without knowing if the traffic signal is at the beginning of the edge or at the end it seems not that useful (edges only have a flag indicating its at the end of the edge). i would suggest you add this to the nodes output instead of the edges and PR that instead!

**Comment by upperwal** [comment]
Thanks @kevinkreiser 

Agree that traffic signals are node properties than edges. As far as I can see from the docs `trace_attributes` has no attribute as `nodes` in the output. Are you suggesting to have a new output called `nodes`?

> without knowing if the traffic signal is at the beginning of the edge or at the end it seems not that useful 

Maybe then instead of a `bool` we can have a float value `0 <= x <= 1` similar to `distance_along_edge` in `matched_point` which signifies distance of traffic signal on an edge. I am assuming that one edge can only have one traffic signal because in real world traffic signals only exist at (or just before) intersections and Valhalla would bifurcate them into two edges. (Two signals without any intersections won't make any sense)

**My Usecase**
I am map-matching public buses GPS traces after which I want to obtain number of junctions, traffic signals and length travelled (till last known GPS point) to get better ETA at each bus stop. Traffic signals play a major role in such estimation.

**Comment by dgearhart** [comment]
@upperwal there are attributes at nodes: https://github.com/valhalla/valhalla/blob/master/docs/api/map-matching/api-reference.md#end-node-items

**Comment by upperwal** [comment]
Thanks @dgearhart 

Adding a `traffic_signal: bool` to `end_node` would only signify that end node is a signal and won't cover the traffic signal somewhere in the middle of an edge. This looks incomplete because a user might want to get all signals while map-matching.

What if we do something like `traffic_signal_distance_along_edge`. Signals at the end would be `1.0`. This looks more like an edge property than a node. What do you say?

**Comment by upperwal** [comment]
@kevinkreiser 

> you will clearly see that the vast majority of traffic signals are at a node

That's true and it makes sense to have a `traffic_signal: bool` within `end_node` attribute. Will work on this and open a PR.

> even if that weren't the case there would be a significant amount of work to add functionality for adding signals to the data that were located x% along an edge (a feature which we will not support at this time)

As the above PR won't support the given case, I would be happy to work on `x% along an edge` use case later on.
<img width="394" alt="Screenshot 2019-09-20 at 2 04 29 PM" src="https://user-images.githubusercontent.com/5246435/65312170-978ef900-dbaf-11e9-9fdc-d4099050e251.png">
